syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package inferencedata;

service RemoteInference {
	rpc Infer (ImageBatch) returns (ResultBatch) {}
}

message Image {
	int32 id = 1;
	bytes image_data = 2;
}

message ImageBatch {
	repeated Image images = 1;
}

message Result {
	int32 id = 1;
	int32 num = 2;
}

message ResultBatch {
	repeated Result results = 1;
}